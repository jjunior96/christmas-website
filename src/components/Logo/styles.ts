import styled, { css } from 'styled-components';

import { LogoProps } from '.';

type ContainerProps = Pick<LogoProps, 'color'>;

export const Container = styled.h2<ContainerProps>`
  ${({ theme, color }) => css`
    color: ${theme.colors[color!]};

    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.bold};
    font-family: 'Pacifico';
  `}
`;
