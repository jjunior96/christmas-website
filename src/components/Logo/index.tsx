import * as S from './styles';

export type LogoProps = {
  color?: 'white' | 'black';
};

const Logo = ({ color = 'white' }: LogoProps) => {
  return <S.Container color={color}>JR</S.Container>;
};

export default Logo;
