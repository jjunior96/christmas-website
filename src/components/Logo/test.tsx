import { screen } from '@testing-library/react';

import { renderWithTheme } from '../../utils/tests/helpers';

import theme from 'styles/theme';

import Logo from '.';

describe('<Logo />', () => {
  it('should render component default', () => {
    const { container } = renderWithTheme(<Logo />);

    expect(screen.getByRole('heading', { name: /jr/i })).toBeInTheDocument();
    expect(screen.getByText(/jr/i)).toHaveStyle({
      color: theme.colors.white
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with color dark', () => {
    renderWithTheme(<Logo color="black" />);

    expect(screen.getByText(/jr/i)).toHaveStyle({
      color: theme.colors.black
    });
  });
});
