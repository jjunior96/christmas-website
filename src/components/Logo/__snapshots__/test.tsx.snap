// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render component default 1`] = `
.c0 {
  color: #fff;
  font-size: 4rem;
  font-weight: 600;
  font-family: 'Pacifico',cursive;
  line-height: 1;
}

<h2
  class="c0"
  color="white"
>
  JR
</h2>
`;
